runtime/src/chpl-const-arg-check.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "chpl-const-arg-check.h"
+
+#include "chplrt.h"
+#include "chpl-linefile-support.h"
+#include "error.h"
+
+// FNV-1a hash function, a "string hash"
+uint64_t const_arg_hash(void* ptr, size_t size) {
+  uint64_t hashValue = 0xcbf29ce484222325; // starting seed
+  uint64_t FNV_prime = 0x100000001b3;
+
+  unsigned char* fakeCharPtr = (unsigned char*)ptr;
```

Comment:
the pointer is real 😄

Reply:
Yeah, but it's not a char pointer, that's why it's fake!